lab2.js

```diff
@@ -86,27 +131,40 @@ var hello = {
 // speak, and method (that you'll place on the prototype) called
 // sayHello.
 
-function SentientBeing () {
+function SentientBeing(homePlanet, language) {
   // TODO: specify a home planet and a language
   // you'll need to add parameters to this constructor
+  this.homePlanet = homePlanet;
+  this.language = language;
 }
 
 // sb is a SentientBeing object
-function sayHello (sb) {
+SentientBeing.prototype.sayHello = function sayHello(sb) {
     // TODO: say hello prints out (console.log's) hello in the
     // language of the speaker, but returns it in the language
     // of the listener (the sb parameter above).
     // use the 'hello' object at the beginning of this exercise
     // to do the translating
+  if (sb === klingon) {
+    console.log(hello['federation standard'] + ' and ' + hello.romulan);
+    return (hello.klingon);
 
-    //TODO: put this on the SentientBeing prototype
+  } else if (sb === romulan) {
+    console.log(hello.klingon + ' and ' + hello['federation standard']);
+    return (hello.romulan);
+
+  } else if (sb === human) {
+    console.log(hello.romulan + ' and ' + hello.klingon);
+    return (hello['federation standard']);
```

Comment:
so you're using bracket notation here to refer to federation standard; you can also use it to access the other properties on hello via passing in a variable. Basically, it's what you and Tina and I were talking about last night.